src/query_parser/parser.y

```diff
@@ -260,7 +266,7 @@ expr(A) ::= expr(B) expr(C) . [AND] {
     }
 }
 
-text_expr(A) ::= text_expr(B) text_expr(C) . [AND] {
+expr(A) ::= text_expr(B) expr(C) . [AND] {
```

Comment:
This is right-recursion - can't we use left-recursion to reduce the stack size?
```expr(A) ::= expr(B) text_expr(C) . [AND] {```

Reply:
It’s not recursion, it’s for when we have shifted on `text_expr` and now have an `expr` we need to reduce with it. The other way around is covered with reduction from `text_expr` to `expr`